jlm/rvsdg/Transformation.cpp

```diff
@@ -52,13 +54,55 @@ TransformationSequence::Run(
   auto statistics = Statistics::Create(rvsdgModule.SourceFilePath().value());
   statistics->StartMeasuring(rvsdgModule.Rvsdg());
 
-  for (const auto & optimization : Transformations_)
+  size_t numPasses = 0;
+  if (DumpRvsdgDotGraphs_)
+  {
+    DumpDotGraphs(
+        rvsdgModule,
+        statisticsCollector.GetSettings().GetOutputDirectory(),
+        "Pristine",
+        numPasses);
+    numPasses++;
+  }
+
+  for (const auto & transformation : Transformations_)
   {
-    optimization->Run(rvsdgModule, statisticsCollector);
+    transformation->Run(rvsdgModule, statisticsCollector);
+    if (DumpRvsdgDotGraphs_)
+    {
+      DumpDotGraphs(
+          rvsdgModule,
+          statisticsCollector.GetSettings().GetOutputDirectory(),
+          "After" + std::string(transformation->GetName()),
+          numPasses);
+      numPasses++;
+    }
   }
 
   statistics->EndMeasuring(rvsdgModule.Rvsdg());
   statisticsCollector.CollectDemandedStatistics(std::move(statistics));
 }
 
+void
+TransformationSequence::DumpDotGraphs(
+    RvsdgModule & rvsdgModule,
+    const util::FilePath & outputDir,
+    const std::string & passName,
+    size_t numPass) const
+{
+  JLM_ASSERT(outputDir.Exists() && outputDir.IsDirectory());
+
+  util::graph::Writer graphWriter;
+  DotWriter_.WriteGraphs(graphWriter, rvsdgModule.Rvsdg().GetRootRegion(), true);
+
+  std::stringstream filePath;
+  filePath << outputDir.to_str() << "/" << std::setw(3) << std::setfill('0') << numPass << "-"
+           << passName << ".dot";
+
+  std::ofstream outputFile;
+  outputFile.open(filePath.str().c_str());
```

Comment:
Do we need `.c_str()` here? `ofstream::open()` can take a `const std::string &`.